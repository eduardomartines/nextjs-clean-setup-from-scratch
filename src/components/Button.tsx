import React from 'react'
import 'tailwindcss/tailwind.css'

export interface ButtonProps {
  primary?: boolean
  label: string
  onClick?: () => void
}

export const Button: React.FC<ButtonProps> = ({ primary = false, label, ...props }) => {
  const mode = primary ? 'bg-blue-500 hover:bg-blue-600' : 'bg-gray-500 hover:bg-gray-600'
  return (
    <button
      className={[
        'focus:outline-none text-white text-sm py-2.5 px-5 rounded-md hover:shadow-lg',
        mode,
      ].join(' ')}
      {...props}
    >
      {label}
    </button>
  )
}
