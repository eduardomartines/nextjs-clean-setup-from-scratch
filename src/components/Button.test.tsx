import renderer from 'react-test-renderer'
import { Button } from './Button'

describe('Button', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Button label="Testing" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
