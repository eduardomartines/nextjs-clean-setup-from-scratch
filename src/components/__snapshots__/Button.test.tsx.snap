// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<button
  className="focus:outline-none text-white text-sm py-2.5 px-5 rounded-md hover:shadow-lg bg-gray-500 hover:bg-gray-600"
>
  Testing
</button>
`;
